src/sql/base/browser/ui/table/plugins/headerFilter.plugin.ts

```diff
@@ -84,7 +85,10 @@ export class HeaderFilter<T extends Slick.SlickData> {
 			.subscribe(this.grid.onClick, (e: DOMEvent) => this.handleBodyMouseDown(e as MouseEvent))
 			.subscribe(this.grid.onColumnsResized, () => this.columnsResized())
 			.subscribe(this.grid.onKeyDown, async (e: DOMEvent) => { await this.handleGridKeyDown(e as KeyboardEvent); });
-		this.grid.setColumns(this.grid.getColumns());
+
+		if (!this.options.shouldSetColumns) {
+			this.grid.setColumns(this.grid.getColumns());
```

Comment:
let me have a look locally whether we actually need this.

Reply:
If the auto size columns plugin is off then the filter menu buttons won't get added if we remove this setColumns call.

Reply:
yes, tested locally, this is actually used to redraw the headers.